sklearn/manifold/_t_sne.py

```diff
@@ -681,12 +688,30 @@ def __init__(self, n_components=2, *, perplexity=30.0,
     def _fit(self, X, skip_num_points=0):
         """Private function to fit the model using X as training data."""
 
+        if self.init == 'warn':
+            # See issue #18018
+            warnings.warn("The default initialization will change from "
+                          "random to PCA in 1.01.", FutureWarning)
+            self.init = 'random'
```

Comment:
We avoid changing attributes that were defined in the init, to avoid unexpected effects when using hyperparameter search. So we prefer defining another variable:
```py
if self.init == 'warn':
    init = "random"
else:
    init = self.init
```

Reply:
Wait, but it is not what the example here https://scikit-learn.org/stable/developers/contributing.html#change-the-default-value-of-a-parameter shows!

Reply:
The example does not overwrite `self.n_clusters`, it defines a new private attribute `self._n_clusters`.

Reply:
Oooh! Indeed. Thanks, I entirely missed it. Pushed a fix now.